#goal definition
# name of the behavior
string behavior

# at what time should we be finished moving/holding
# holding only meaningful if blocking or return to prior is true
time end_move
time end_hold

# return to previous pose when behavior is done?
bool return_to_prior

# if the relevant DOFs are in conflict, what to do?
# Queue: play behavior once DOFs are free; end_move and end_hold are taken to be relative to time.now
# Drop: never play the behavior
# Override: cancel the current behavior and play the new one
int32 if_conflict
int32 QUEUE = 0
int32 DROP = 1
int32 OVERRIDE = 2

# some behaviors may take arguments, e.g., a frame for lookat
string[] args

# have robot controller block on this
# if true, then wait_for_result will block until the behavior is done
bool wait_and_block
---
#result definition
int32 result
int32 DONE = 1
int32 FAILED = 2
---
#feedback
# the name of the behavior we're waiting on
int32 status
int32 WAITING = 1
int32 PLAYING = 2
